doc/tools/make_rst.py

```diff
@@ -1059,10 +1061,16 @@ def make_rst_class(class_def: ClassDef, state: State, dry_run: bool, output_dir:
                     f.write("----\n\n")
 
                 if i == 0:
-                    f.write(
-                        f".. _class_{class_name}_operator_{sanitize_operator_name(m.name, state)}_{m.return_type.type_name}:\n\n"
-                    )
-
+                    if len(m.parameters) > 0:
+                        out = f".. _class_{class_name}_operator_{sanitize_operator_name(m.name, state)}_{m.return_type.type_name}"
+                        for parameter in m.parameters:
+                            out += f"_{parameter.type_name.type_name}"
+                        out += f":\n\n"
+                        f.write(out)
+                    else:
```

Comment:
There is not much sense in branching like this. You can just leave the first part, it would already handle not having any parameters just fine.